environments/openshift/prometheus.libsonnet

```diff
@@ -0,0 +1,271 @@
+        role.mixin.metadata.withNamespace(namespace) +
+        role.withRules(coreRule);
+
+      local roleList = k3.rbac.v1.roleList;
+      roleList.new([newSpecificRole(x) for x in $._config.ams.prometheus.namespaces]),
+    prometheus:
+      local statefulSet = k.apps.v1.statefulSet;
+      local container = statefulSet.mixin.spec.template.spec.containersType;
+      local resourceRequirements = container.mixin.resourcesType;
+      local selector = statefulSet.mixin.spec.selectorType;
+      local deployment = k.apps.v1.deployment;
+      local container = deployment.mixin.spec.template.spec.containersType;
+      local volume = k.apps.v1beta2.statefulSet.mixin.spec.template.spec.volumesType;
+      local volumeMount = container.volumeMountsType;
+
+      local resources =
+        resourceRequirements.new() +
+        resourceRequirements.withLimits($._config.ams.resourceLimits) +
+        resourceRequirements.withRequests($._config.ams.resourceRequests);
+
+      {
+        apiVersion: 'monitoring.coreos.com/v1',
+        kind: 'Prometheus',
+        metadata: {
+          name: $._config.ams.prometheus.name,
+          namespace: $._config.namespace,
+          labels: {
+            prometheus: $._config.ams.prometheus.name,
+          },
+        },
+        spec: {
+          replicas: $._config.ams.prometheus.replicas,
+          version: $._config.versions.prometheus,
+          baseImage: $._config.imageRepos.prometheus,
+          serviceAccountName: 'prometheus-' + $._config.ams.prometheus.name,
+          serviceMonitorSelector: {},
+          podMonitorSelector: {},
+          serviceMonitorNamespaceSelector: {},
+          nodeSelector: { 'kubernetes.io/os': 'linux' },
+          ruleSelector: selector.withMatchLabels({
+            role: 'alert-rules',
+            prometheus: $._config.ams.prometheus.name,
+          }),
+          resources: resources,
+          securityContext: {
+            runAsUser: 1000,
+            runAsNonRoot: true,
+            fsGroup: 2000,
+          },
+          remoteWrite: $._config.ams.prometheus.remoteWrite,
+          containers: [
+            container.new('remote-write-proxy', '%s:%s' % [$._config.imageRepos.remoteWriteProxy, $._config.versions.remoteWriteProxy]) +
+            container.withArgs([]) +
+            container.withPorts([{ name: 'http', containerPort: $._config.ams.proxyPort }]) +
+            container.withVolumeMounts([volumeMount.new('prometheus-%s' % $.prometheusAms.configmap.metadata.name, '/nginx.conf', true)]),
+          ],
+          volumes+: [
+            volume.fromConfigMap('prometheus-%s' % $.prometheusAms.configmap.metadata.name, $.prometheusAms.configmap.metadata.name, ['nginx.conf']),
+          ],
+        },
+      },
+    serviceMonitor:
+      {
+        apiVersion: 'monitoring.coreos.com/v1',
+        kind: 'ServiceMonitor',
+        metadata: {
+          name: 'prometheus-ams',
```

Comment:
this looks like the only field in jsonnet that is explicitly assigned the string `'prometheus-ams'`, instead of `'prometheus' + $._config.ams.prometheus.name`